pkg/backporter/backporter.go

```diff
@@ -278,94 +304,218 @@ func GetBugHandler(client bugzilla.Client) HandlerFuncWithErrorReturn {
 	}
 }
 
+func getClonesTemplateData(bugID int, w http.ResponseWriter, client bugzilla.Client, allTargetVersions sets.String) (*ClonesTemplateData, error) {
+	bug, err := client.GetBug(bugID)
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		wpErr := writePage(w, "Not Found", errorTemplate, fmt.Sprintf("Bug#%d not found", bugID))
+		if wpErr != nil {
+			http.Error(w, "Error building page.", http.StatusInternalServerError)
```

Comment:
Not sure if go catches this, but I doubt its valid to send the same header multiple times and you already send it on line 310.

Comment applies to all occurences of this.

Reply:
I think it should since we are setting it to 200 here (https://github.com/openshift/ci-tools/blob/master/cmd/bugzilla-backporter/main.go#L77) and the last applied header seems to take effect

Reply:
I think the traceResponseWriter only uses it for metrics because the golang http server will send a http 200 status if you do not call `WriterHeader`. Also:

```
go doc net/http.ResponseWriter.WriteHeader
[...]
// Only one header may be written.
[...]
```

And even if that wasn't the case I would argue that this must always be a 404. We do not want to confuse clients with a javalike `exception handling exception`, we could write that into the body but we should not mess with the http status if that happens

Reply:
Also looking at the `traceResponseWriter` it doesn't do what its supposed to do and will always increase the metric for the statusCode it was created with rather even if `WriteHeader` was called because it doesn't implement a `WriteHeader` wrapper like here: https://github.com/openshift/ci-tools/blob/2f2c559688e4a27349bae36bf1e38253f52930cc/cmd/ci-operator-configresolver/main.go#L48

Could you please create a different PR that moves the `traceResponseWriter` from `cmd/ci-operator-configresolver` into `pkg/httphelper` and re-uses that in both the `ci-operator-configresolver` and the `bugzilla-backporter` so we don't end up with duplicated code?

Reply:
Ok